src/api/app/controllers/webui/package_controller.rb

```diff
@@ -246,7 +246,7 @@ def submit_request_dialog
   end
 
   def submit_request
-    required_parameters :project, :package
+    required_parameters :project, :package, :targetproject
```

Comment:
The target project should still be optional, since it is by default the link target. Except when you fill this in the webui by default. Do you?

Small side note: There seems to be code missing in webui to change a submit request to a maintenance request, if the target is a maintenance_release project. osc is doing automatically so when it gets a "submit_request_rejected" error on creating the request, but the webui could be more clever since it has all the information by default.

This is more then you want to achieve with this patch, but we should IMHO not overwrite the automatic target detection in the api code.

Is something else broken when you do this without a targetproject? (on a linked package)